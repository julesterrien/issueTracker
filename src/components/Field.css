.field-wrapper {
  display: flex;
  flex-direction: column;
}

.field-label {
  margin-bottom: 0.5rem;
}

.field-input-wrapper {
  display: flex;
  flex-direction: column;
}

.field-input {
  padding: 0.5rem 0.3rem;
  border: 1px solid transparent;
  outline: none;
}

.field-input:focus {
  border-bottom: 2px solid black;
}

.field-error {
  font-size: 0.825rem;
  color: red;
}
