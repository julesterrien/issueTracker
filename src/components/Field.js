import React from 'react';

const Field = ({
  label,
  name,
  type = 'text',
  placeholder,
  input,
  autoFocus = false,
  meta: {
    touched,
    error,
  }
}) => {
  return (
    <div>
      <label htmlFor={name}>
        {label}
      </label>
      <div>
        <input {...input} autoFocus={autoFocus} placeholder={placeholder} name={name} type={type} />
        {touched && (error && <span>{error}</span>)}
      </div>
    </div>
  );
};

export default Field;
