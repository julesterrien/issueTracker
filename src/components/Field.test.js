import React from 'react';
import renderer from 'react-test-renderer';

import Field from './Field';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Field
        name="fieldName"
        type="text"
        placeholder="placeholder"
        input={{}}
        autoFocus={false}
        meta={{
          touched: true,
          error: true,
        }}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
