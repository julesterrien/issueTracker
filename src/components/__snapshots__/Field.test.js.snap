// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="field-wrapper"
>
  <label
    className="field-label"
    htmlFor="fieldName"
  />
  <div
    className="field-input-wrapper"
  >
    <input
      autoFocus={false}
      className="field-input"
      name="fieldName"
      placeholder="placeholder"
      type="text"
    />
    <span
      className="field-error"
    />
  </div>
</div>
`;
